src/components/FullscreenHandler/index.js

```diff
@@ -0,0 +1,50 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import { PROP_TYPE_CHILDREN } from '../constants'
+import { renderChildren } from '../helpers'
+
+const fullscreenEvents = [
+  'fullscreenchange',
+  'mozfullscreenchange',
+  'webkitfullscreenchange',
+  'msfullscreenchange',
+]
+
+export default class FullscreenHandler extends PureComponent {
+  static propTypes = {
+    onFullscreenChange: PropTypes.func,
+    children: PROP_TYPE_CHILDREN.isRequired,
+  }
+
+  state = {
+    fullscreenElement: undefined,
```

Comment:
the default value for `document.fullscreenElement` when not fullscreen is `null`.  not that it matters a whole lot, but we should probably keep with that, as when you move from FS to normal, we will be setting this value to `null`